torch/quantization/observer.py

```diff
@@ -394,10 +394,12 @@ def forward(self, x_orig):
         x = x_orig.detach()  # avoid keeping autograd tape
         x = x.to(self.min_val.dtype)
         min_val_cur, max_val_cur = torch._aminmax(x)
+        self.min_val = self.min_val.to(x.device)
+        self.max_val = self.max_val.to(x.device)
```

Comment:
does the `DataParallel` test past with this?  Would something like

```
self.min_val.copy_(min_val.to(x.device))
```

work instead?

Reply:
Yes the tests in `TestDistributed` passed with this. 
MinMaxObserver is a little different and needs the `.to` here since we call `min_val = torch.min(min_val_cur, self.min_val)` after this which will fail since `self.min_val` and `min_val_cur` are on different devices.

Reply:
> self.min_val and min_val_cur are on different devices.

do we know why this happens, btw?  If the model owner does

```
inputs = inputs.cuda()
model = model.cuda()
model(inputs)
```

we would expect that all observes in model are already on the right device.

btw, it doesn't need to block this PR, I'm just wondering about the root cause for future improvements.

Reply:
I believe it mainly happens when we use the observers standalone and they are not part of a model. Not sure if that is a use-case we want to support.
We can add a device argument to observers, which will ensure all the buffers registered in the observer are on the expected device (i.e. match the device of the input it is run on).

Reply:
> I believe it mainly happens when we use the observers standalone and they are not part of a model. Not sure if that is a use-case we want to support.

> Previosuly the scale and zero_point were returned on the CPU even if

Hmm, sorry, so is the goal of the PR to support this standalone use case?  I.e. the user created the observer on CPU, passes a CUDA Tensor, and this is fixing the crash?  If yes, it might be worth deciding if this should in fact be supported, or if we should throw a useful exception and ask the user to move the observer to the right device.